zerver/views/users.py

```diff
@@ -466,10 +467,17 @@ def get_members_backend(request: HttpRequest, user_profile: UserProfile, user_id
         # If email addresses are only available to administrators,
         # clients cannot compute gravatars, so we force-set it to false.
         client_gravatar = False
+
+    # If user data of a single user is asked for via
+    # 'GET /users/{user_id}' or 'GET /users/{email},
+    # we check whether we have a 'user_id' or an 'email'.
```

Comment:
I'd explain the why part rather than what we're doing -- that the `user_id` and `email` parameters are defined in `urls.py`, not via `REQ`.  Though we might be able to validate these with `REQ` and the `in_path` flag, and that'll likely be better for doing the API documentation?  Not sure.

Reply:
@timabbot We can't use 'path_only' because it requires them to be always passed to the view then.